### PR TITLE
[addons] fixes/cleanup

### PR DESCRIPTION
**Pull request by tamland**
Some followup fixes for #9051

### CODE REVIEW

File: xbmc/addons/AddonDatabase.cpp [hunk 1]
```diff
@@ -310,13 +310,13 @@ int CAddonDatabase::AddAddon(const AddonPtr& addon,
     if (NULL == m_pDS.get()) return -1;
 
     std::string sql = PrepareSQL("insert into addon (id, type, name, summary,"
-                               "description, stars, path, icon, changelog, "
+                               "description, path, icon, changelog, "
                                "fanart, addonID, version, author, disclaimer, minversion)"
                                " values(NULL, '%s', '%s', '%s', '%s', %i,"
-                               "'%s', '%s', '%s', '%s', '%s','%s','%s','%s','%s')",
```
Comment (anaconda) [inline xbmc/addons/AddonDatabase.cpp]: Oops

File: xbmc/addons/AddonManager.cpp [hunk 2]
```diff
@@ -170,9 +167,12 @@ void CAddonMgr::FillCpluffMetadata(const cp_plugin_info_t* plugin, CAddonBuilder
     ADDONDEPS dependencies;
     for (unsigned int i = 0; i < plugin->num_imports; ++i)
     {
-      if (plugin->imports[i].plugin_id && plugin->imports[i].version)
-        dependencies.insert(std::make_pair(std::string(plugin->imports[i].plugin_id),
-            std::make_pair(AddonVersion(plugin->imports[i].version), plugin->imports[i].optional != 0)));
+      if (plugin->imports[i].plugin_id)
+      {
+        std::string id(plugin->imports[i].plugin_id);
+        AddonVersion version(plugin->imports[i].version ? plugin->imports[i].version : "0.0.0");
```
Comment (anaconda) [inline xbmc/addons/AddonManager.cpp]: Btw wiki says `version` is required. I'd print a warning there.

File: xbmc/addons/AddonDatabase.cpp (hunk 1 above)
Comment (tamland) [inline xbmc/addons/AddonDatabase.cpp]: Thanks for the save!

File: xbmc/addons/AddonManager.cpp (hunk 2 above)
Comment (tamland) [inline xbmc/addons/AddonManager.cpp]: Unfortunately that will cause the same spamming as f2555f21d7 during repo update

Comment (MartijnKaijser) [inline xbmc/addons/AddonManager.cpp]: We will just enforce it for official repo. What the rest does is their problem

Comment (tamland) [inline xbmc/addons/AddonManager.cpp]: To be clear: I don't see any technical problems with having version optional and implicitly defined as "0.0.0". We already do that for minversion, among other things

**Comment by tamland** [comment]
jenkins build this please

**Pull request merged**